ibis/expr/types/relations.py

```diff
@@ -2391,17 +2403,20 @@ def rename(name):
         else:
             rename = method
 
-        exprs = []
+        exprs = {}
+        fields = self.op().fields
+        rename_is_none = rename is None
         for c in self.columns:
-            if c in renamed:
-                expr = renamed[c]
+            if (new_name_op := renamed.get(c)) is not None:
+                new_name, op = new_name_op
             else:
-                expr = self[c]
-                if (name := rename(c)) is not None:
-                    expr = expr.name(name)
-            exprs.append(expr)
+                op = fields[c]
+                if rename_is_none or (new_name := rename(c)) is None:
```

Comment:
```suggestion
                if rename is None or (new_name := rename(c)) is None:
```

Any reason this isn't inlined?

Reply:
I think it was to avoid recomputing the thing. It's probably undetectable from a performance perspective, we've got other problems if a pointer comparison is what our performance hinges on 😬.

Inlining it :)